### ISSUE TITLE
AbortionError(code=StatusCode.FAILED_PRECONDITION, details="Attempting to use un initialized value FeatureExtractor/MobilenetV1/Conv2d_0/weights          [[Node: FeatureExtractor/MobilenetV1/Conv2d_0/weights/read = Identity[T=DT_FLOAT, _class=["loc:@FeatureExtractor/Mobilenet V1/Conv2d_0/weights"], _output_shapes=[[3,3,3,32]], _device="/job:localhost/replica:0/task:0/cpu:0"](FeatureExtractor/MobilenetV1/C onv2d_0/weights)]]")

### ISSUE THREAD

**Issue by Raj-08** [tensorflow/serving#725]
This is the code for create_proto file: 

   `def preprocess_image(image_buffer):

    image = tf.image.decode_jpeg(image_buffer, channels=3)

    image = tf.expand_dims(image, dim=0) # Add one batch dimension.
    return image

    checkpoint_dir = '../training/inference_graph/inference_graph_v3/inference_graph_90120/model.ckpt'

    PATH_TO_LABELS='../training/object_detection.pbtxt'
    NUM_CLASSES = 2
    output_dir = '/home/ubuntu/gfav/multiple_object/object_detection_serving/object_detection_export'
    model_version = '1'
    detection_graph = tf.Graph()
    config = tf.ConfigProto()
    config.gpu_options.allow_growth = True
    with detection_graph.as_default():
    saver = tf.train.import_meta_graph(checkpoint_dir + ".meta")
    with tf.Session(graph=detection_graph, config=config) as sess:
        all_variables = tf.get_collection_ref(tf.GraphKeys.GLOBAL_VARIABLES)
        sess.run(tf.variables_initializer(all_variables))     
        tf.global_variables_initializer().run()
        tf.tables_initializer().run()
        saver.restore(sess, checkpoint_dir)    
        image_tensor = detection_graph.get_tensor_by_name('image_tensor:0')
        boxes = detection_graph.get_tensor_by_name('detection_boxes:0')
        scores = detection_graph.get_tensor_by_name('detection_scores:0')
        classes = detection_graph.get_tensor_by_name('detection_classes:0')
        export_path = os.path.join(
            tf.compat.as_bytes(output_dir),
            tf.compat.as_bytes(str(model_version)))
        if os.path.exists(export_path):
            shutil.rmtree(export_path)
        # create model builder
        builder = tf.saved_model.builder.SavedModelBuilder(export_path)
        predict_tensor_inputs_info = \
            tf.saved_model.utils.build_tensor_info(image_tensor) 
        predict_tensor_scores_info = \
        tf.saved_model.utils.build_tensor_info(scores)
        predict_tensor_classes_info = \
        tf.saved_model.utils.build_tensor_info(classes)
        predict_tensor_boxes_info = \
        tf.saved_model.utils.build_tensor_info(boxes)    
        prediction_signature = (
        tf.saved_model.signature_def_utils.build_signature_def( 
            inputs={'images': predict_tensor_inputs_info}, 
            outputs={'boxes': predict_tensor_boxes_info, \
                     'scores': predict_tensor_scores_info, \
                     'classes': predict_tensor_classes_info},
            method_name=\
                tf.saved_model.signature_constants.PREDICT_METHOD_NAME)
        )       
        legacy_init_op = tf.group(tf.tables_initializer(), name='legacy_init_op')
        builder.add_meta_graph_and_variables(
            sess, [tf.saved_model.tag_constants.SERVING],
            signature_def_map={
                'detection_signature': prediction_signature
            },
            assets_collection=tf.get_collection(tf.GraphKeys.ASSET_FILEPATHS),
    legacy_init_op=legacy_init_op)
        builder.save()

        #####This is the code for grpc client : ########3
      tensor_server = '172.17.0.3:9000'
      host, port = tensor_server.split(':')
      channel = implementations.insecure_channel(host, int(port))
      stub = prediction_service_pb2.beta_create_PredictionService_stub(channel)
      open('JPEGImages/7117.jpg', 'rb') as f: 
      import cv2
      i = cv2.imread('JPEGImages/7117.jpg') 
      request = predict_pb2.PredictRequest()
      request.model_spec.name = 'object_detection'
      request.model_spec.signature_name = "detection_signature"
      request.inputs['images'].CopyFrom(
      tf.contrib.util.make_tensor_proto(i, shape=i.shape))
      result = stub.Predict(request, 60.0)
      category_index = load_label_map(FLAGS.path_to_labels)
      boxes = result.outputs['boxes'].float_val
      classes = result.outputs['classes'].float_val
      scores = result.outputs['scores'].float_val`

I run the create_proto file ,which succesfully saves the variables and model.pb file , but when i run the grpc client i run into the mentioned error.

"AbortionError(code=StatusCode.FAILED_PRECONDITION, details="Attempting to use un initialized value FeatureExtractor/MobilenetV1/Conv2d_0/weights          [[Node: FeatureExtractor/MobilenetV1/Conv2d_0/weights/read = Identity[T=DT_FLOAT, _class=["loc:@FeatureExtractor/Mobilenet V1/Conv2d_0/weights"], _output_shapes=[[3,3,3,32]], _device="/job:localhost/replica:0/task:0/cpu:0"](FeatureExtractor/MobilenetV1/C onv2d_0/weights)]]")"

**Comment by Raj-08** [comment]
Was a tensorflow version issue
